src/Subscriptions.ts

```diff
@@ -68,28 +57,46 @@ export function handleFeeAmountChanged(event: FeeAmountChanged): void {
   subscriptions.save()
 }
 
-export function handlePrePaymentPeriodsChanged(event: PrePaymentPeriodsChanged): void {
-  let subscriptions = SubscriptionModule.load(event.address.toHex())
-  subscriptions.prePaymentPeriods = event.params.currentPrePaymentPeriods
-  subscriptions.save()
-}
-
 export function handleGovernorSharePctChanged(event: GovernorSharePctChanged): void {
   let subscriptions = SubscriptionModule.load(event.address.toHex())
   subscriptions.governorSharePct = BigInt.fromI32(event.params.currentGovernorSharePct)
   subscriptions.save()
 }
 
-export function handleLatePaymentPenaltyPctChanged(event: LatePaymentPenaltyPctChanged): void {
-  let subscriptions = SubscriptionModule.load(event.address.toHex())
-  subscriptions.latePaymentPenaltyPct = BigInt.fromI32(event.params.currentLatePaymentPenaltyPct)
-  subscriptions.save()
+export function handleAppFeeSet(event: AppFeeSet): void {
+  let appFee = loadOrCreateAppFee(event.params.appId)
+  appFee.isSet = true
+  appFee.amount = event.params.amount
+  appFee.instance = event.address.toHex()
+  appFee.save()
 }
 
-export function handleResumePenaltiesChanged(event: ResumePenaltiesChanged): void {
-  let subscriptions = SubscriptionModule.load(event.address.toHex())
-  subscriptions.resumePrePaidPeriods = event.params.currentResumePrePaidPeriods
-  subscriptions.save()
+export function handleAppFeeUnset(event: AppFeeUnset): void {
+  let appFee = AppFee.load(event.params.appId.toString())
```

Comment:
Why not simply using `loadOrCreateAppFee`?

Reply:
See response above. Here it should never occur that `appFee` becomes null either.